app/controllers/courses_controller.rb

```diff
@@ -5,15 +5,18 @@ class CoursesController < ApplicationController
   before_action :build_courses, only: %i[index about requirements fees salary]
   before_action :build_course, except: %i[index preview]
   before_action :build_course_for_preview, only: :preview
-  before_action :build_provider, except: %i[index confirmation]
```

Comment:
Why remove the action?

Reply:
Ah, I see that `build_provider` is essentially being replaced by `build_provider_from_provider_code`. I think we should use `build_provider_from_provider_code` as a `before_action` ... there's validity to being explicit and running that in the actions that need it, but we _are_ using `before_actions` in our controllers, so should keep consistent.